testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -1263,6 +1263,41 @@ public void testInsertDeletUpdateWithPrestoAndHive()
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
+    public void testDeletePartitionedTable()
+    {
+        withTemporaryTable("delete_partitioned", true, true, NONE, tableName -> {
+            onPresto().executeQuery(format("CREATE TABLE %s WITH (transactional = true, partitioned_by = ARRAY['regionkey'])" +
+                    " AS SELECT nationkey, name, regionkey FROM tpch.tiny.nation", tableName));
+
+            verifySelectForPrestoAndHive("SELECT COUNT(*) FROM " + tableName, "TRUE", row(25));
+
+            onPresto().executeQuery(format("DELETE FROM %s WHERE regionkey = 4 AND nationkey %% 10 = 3", tableName));
+            verifySelectForPrestoAndHive("SELECT COUNT(*) FROM " + tableName, "TRUE", row(24));
+
+            onPresto().executeQuery(format("DELETE FROM %s WHERE regionkey = 4", tableName));
+            verifySelectForPrestoAndHive("SELECT COUNT(*) FROM " + tableName, "TRUE", row(20));
```

Comment:
You can add a comment here
```
// TODO (https://github.com/trinodb/trino/issues/7291) this currently removes partition directories, should create delete_delta files
```

(optional)

Reply:
I'll leave this alone, since that bug is not specific to this test